nanoFramework.Tools.DebugLibrary.Shared/PortSerial/DeviceWatcher.cs

```diff
@@ -86,7 +88,125 @@ public void Start()
             }
         }
 
-        private List<string> GetPortNames()
+        public List<string> GetPortNames()
+        {
+
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? GetPortNames_Linux()
+                : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? GetPortNames_OSX()
+                : RuntimeInformation.IsOSPlatform(OSPlatform.Create("FREEBSD")) ? GetPortNames_FreeBSD()
+                : GetPortNames_Windows();
+        }
+
+        private List<string> GetPortNames_Linux()
+        {
+            const string sysTtyDir = "/sys/class/tty";
+            const string sysUsbDir = "/sys/bus/usb-serial/devices/";
+            const string devDir = "/dev/";
+
+            if (Directory.Exists(sysTtyDir))
+            {
+                // /sys is mounted. Let's explore tty class and pick active nodes.
+                List<string> ports = new List<string>();
+                DirectoryInfo di = new DirectoryInfo(sysTtyDir);
+                var entries = di.EnumerateFileSystemInfos(@"*", SearchOption.TopDirectoryOnly);
+                foreach (var entry in entries)
+                {
+                    // /sys/class/tty contains some bogus entries such as console, tty
+                    // and a lot of bogus ttyS* entries mixed with correct ones.
+                    // console and tty can be filtered out by checking for presence of device/tty
+                    // ttyS entries pass this check but those can be filtered out
+                    // by checking for presence of device/id or device/of_node
+                    // checking for that for non-ttyS entries is incorrect as some uart
+                    // devices are incorrectly filtered out
+                    bool isTtyS = entry.Name.StartsWith("ttyS", StringComparison.Ordinal);
```

Comment:
Maybe a good idea to compare it with OrdinalIgnoreCase?

Reply:
Linux is case sensitive, so all good like that. It's expected to be lower case.